lib/rule-tester/rule-tester.js

```diff
@@ -1258,7 +1280,11 @@ class RuleTester {
                         this.constructor[valid.only ? "itOnly" : "it"](
                             sanitize(typeof valid === "object" ? valid.name || valid.code : valid),
                             () => {
-                                testValidTemplate(valid);
+                                try {
+                                    testValidTemplate(valid);
+                                } finally {
+                                    runHook(valid, "after");
+                                }
```

Comment:
this approach allows to run the `after()` hook anyway.
errors thrown by `before()` and by the test case itself are not being catched, so that the workflow operation remains intact.